### ISSUE TITLE
Follow-up after 14 days do not drop out of the follow-up list 

### ISSUE THREAD

**Issue by foyiri** [hzi-braunschweig/SORMAS-Project#1934]
Bug Description
Follow-up covid-19 cases after 14 days do not drop out of the follow-up list, am sure it still calculating the 2-21 days incubation period

Steps to Reproduce
1. It is expected that after the contact tracers have completed the 14 days of visit to the contact of an index case it should be move to follow-up completed  but this is not happening right now

**Comment by Iheanacho2027** [comment]
Yes this is very important at @MateStrysewskeSym @MartinWahnschaffeSymeda. I have logged this previously on #1885

**Comment by bernardsilenou** [comment]
My comment on April 23, 2020 for issue #1648 might be a solution to this. 
@foyiri  This is not a bug, we just need to redefine the logic and nomenclature of the contact status,
@MartinWahnschaffeSymeda can we discuss this in the days to come since it is part of the new sprint?

**Comment by Iheanacho2027** [comment]
Pls consider my comment in #1885

**Comment by MateStrysewskeSym** [comment]
@foyiri @Iheanacho2027 Please confirm that this is happening for contacts that **do have a **cooperative visit on the last day of follow-up**. If not, this is expected behaviour.

**Comment by MartinWahnschaffeSymeda** [comment]
Addition: If they don't have a visit on the last day and the follow-up should still end, we need to define the rules for this. How long should follow-up automatically be extended when no last visit is entered?

**Comment by MateStrysewskeSym** [comment]
I've actually been able to find the source of this bug. The logic that automatically sets the follow-up status to completed when the last visit is done AFTER the original follow-up period just does not work. This will be fixed alongside #1329.

**Comment by Iheanacho2027** [comment]
Ohk @MateStrysewskeSym whether there was a cooperative visit or not, its expected that follow up should end either 14 days for COVID-19 or 21 days for lassa from inputted date of last contact as its the incubation period for both diseases respectively. So if the contact isnt converted to a case within that time period with respect to the date of last contact, it shouldnt remain active but should be dropped @Chinedar @bernardsilenou

**Comment by MateStrysewskeSym** [comment]
@Iheanacho2027 A contact can't be active and dropped at the same time, at least right now. Both are mutually exclusive states.

I'd like to hear the opinions of @Chinedar and @bernardsilenou on this as well because my personal feeling is that dropping contacts automatically should never be done as long as there's the risk of the contact becoming symptomatic, and that's definitely the case if the follow-up has not been completed.

**Comment by bernardsilenou** [comment]
> nted this logic for this exact reason - because there needs to be someone who confirms that the contact has not become symptomatic, and that is only definitely possible when the last follow-up visit has been done.

**Comment by bernardsilenou** [comment]
> @Iheanacho2027 A contact can't be active and dropped at the same time, at least right now. Both are mutually exclusive states.

I agree with this also

> I'd like to hear the opinions of @Chinedar and @bernardsilenou on this as well because my personal feeling is that dropping contacts automatically should never be done as long as there's the risk of the contact becoming symptomatic, and that's definitely the case if the follow-up has not been completed.
I agree also, dropped/discarded contacts should not depend on their follow up status but on reason that "de-qualify" them as being contacts: eg source case is not a case. These contacts will also not count in any statistics or percentage related to contacts. However, we need validation between contact status and followup status: eg 
dropped ==> canceled followup
not a contact ==> dropped, etc
But not always vice versa. I will create another issue with all the validation rules and we can discuss further on them.

> We implemented this logic for this exact reason - because there needs to be someone who confirms that the contact has not become symptomatic, and that is only definitely possible when the last follow-up visit has been done.
@MateStrysewskeSym remember we had a discussion about this point and thought of having a category for contact status whose last date of followup is over due but no data is recorded to tell us if they converted to case or are dropped. Can we have a category for contact status  called
"not converted to case" (if expected followup is overdue ie NOT {active or converted to case or discarded})

**Comment by bernardsilenou** [comment]
Just wish to add some definitions of the categories of contact status followup status and contact classification. Please correct any point that I may have missed.

1**contact status**
1.1 active: expected followup is in future of the present date, automatically defined based on last date of can tact with case
1.2 converted to case: 
1.2 discarded/dropped: contact do no longer meet the criteria to be considered as a contact, eg case is no longer a case, case is dropped, deleted, contact classification is not a contact, etc
1.4 not converted to case: (if expected followup is overdue ie NOT {active or converted to case or discarded})
	
2 **followup status** (dropped contacts do not contribute to these counts and their proportions)	
2.1 under followup: active contacts
2.2 canceled followup : pulled from the set of  all contacts status except "discarded" 
2.3 lost to followup :pulled from the set of  all contacts status except "discarded" 
2.4 overdue followup: pulled from the set of  all contacts status except "active and discarded" 

3 **contact classification**	
    unconfirmed
    confirmed
    not a contact

**Comment by Iheanacho2027** [comment]
@bernardsilenou i guess @MateStrysewskeSym misunderstood my comment. I didnt say a contact can be active and dropped at the same time. I said "if the contact isnt converted to a case within that time period with respect to the date of last contact, it should not remain active, and should be dropped"

**Comment by bernardsilenou** [comment]
@Iheanacho2027 discarded/dropped means the entity (contact) in this case is excluded from the universal set of the entity. You mentioned that "if the contact is not converted to a case within that time period with respect to the date of last contact, it should not remain active, but should be dropped", this is not good to do this because a contact can be visited fro the first week and not for the second week, so we do not need to drop the contact. In my comment above i mentioned adding another category called "not converted to case".

In summary, dropping a contact should not depend on the visit history of the contact. However, the visit can depend on the contact status, e,g if a contact is classified as dropped, then visit should be "canceled followup"

**Comment by MateStrysewskeSym** [comment]
Further discussion of the contact status refinement: #1855
